src/program.ts

```diff
@@ -3626,10 +3659,17 @@ export class Class extends TypedElement {
     }
     return false;
   }
+
+  addImplementer(_class: Class): void {
+    this.implementers.add(_class);
+    if (this.base != null) {
+      (<Class>this.base).addImplementer(_class);
```

Comment:
What if base is null, should that maybe hit an assertion?

Reply:
Actually this should just be when the base is no longer a class.